apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -381,35 +398,47 @@ public Builder addApplicationInterceptor(@Nonnull ApolloInterceptor interceptor)
      * @return The configured {@link ApolloClient}
      */
     public ApolloClient build() {
-      checkNotNull(okHttpClient, "okHttpClient is null");
       checkNotNull(serverUrl, "serverUrl is null");
 
-      apolloLogger = new ApolloLogger(logger);
-      moshi = moshiBuilder.build();
+      ApolloLogger apolloLogger = new ApolloLogger(logger);
+      Moshi moshi = moshiBuilder.build();
 
+      okhttp3.Call.Factory callFactory = this.callFactory;
+      boolean localClient = this.callFactory == null;
+      if (localClient) {
+        callFactory = new OkHttpClient();
+      }
+
```

Comment:
Can we just check if user provided callFactory is instanceof OkHttpClient then add interceptor otherwise just ignore and log warning that http cache won't be enabled unless user adds it by himself.